test/e2e/validation_test.go

```diff
@@ -273,3 +274,38 @@ func createWindowsMachineConfig(namespace string, isReplicasFieldRequired bool,
 	}
 	return wmc
 }
+
+// testWorkerLabel tests if the worker label has been applied properly
+func testWorkerLabel(t *testing.T, nodeCount int) {
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup()
+	nodes, err := framework.Global.KubeClient.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: wmc.WindowsOSLabel})
+	require.NoError(t, err, "error querying the nodes")
+	require.Equal(t, len(nodes.Items), nodeCount, "mismatched node count")
+	for _, node := range nodes.Items {
+		assert.Contains(t, node.Labels, nc.WorkerLabel, "expected node label to be present")
+	}
+}
+
+// testNodeTaint tests if the Windows node has the Windows taint
+func testNodeTaint(t *testing.T) {
+	// windowsTaint is the taint that needs to be applied to the Windows node
+	windowsTaint := corev1.Taint{
+		Key:    "os",
+		Value:  "Windows",
+		Effect: corev1.TaintEffectNoSchedule,
+	}
+	nodes, err := framework.Global.KubeClient.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: wmc.WindowsOSLabel})
```

Comment:
You could have just setup the context which might already have the information related to nodes.